job-dsl-plugin/src/main/groovy/javaposse/jobdsl/plugin/ExecuteDslScripts.java

```diff
@@ -366,8 +390,10 @@ private void updateGeneratedJobMap(AbstractProject<?, ?> seedJob, Set<GeneratedJ
 
                 SeedReference oldSeedReference = generatedJobMap.get(item.getFullName());
                 if (!newSeedReference.equals(oldSeedReference)) {
-                    generatedJobMap.put(item.getFullName(), newSeedReference);
```

Comment:
in case of a dry run, the whole updateGeneratedJobMap method has no effect, so it can be skipped completely.